pkg/api/v1/types.go

```diff
@@ -1806,8 +1811,9 @@ type PodSpec struct {
 	// Value must be a positive integer.
 	ActiveDeadlineSeconds *int64 `json:"activeDeadlineSeconds,omitempty" protobuf:"varint,5,opt,name=activeDeadlineSeconds"`
 	// Set DNS policy for containers within the pod.
-	// One of 'ClusterFirst' or 'Default'.
-	// Defaults to "ClusterFirst".
+	// One of 'ClusterFirstAlways', 'ClusterFirst' or 'Default'.
+	// Defaults to "ClusterFirst" unless hostNetwork is true, otherwise Default.
```

Comment:
The way you wrote this got me thinking.  The *right* way to do this would be to literally make the default value be changed if this field is not specified AND the pod is using hostnetwork.  That way, if this WAS specified, we could just respect it.  No need to add a new enum.

But this is a breaking change in the strictest sense, and it would potentially impact existing pods by changing their behavior on restart.  probably nobody cares, but I can't justify the break.  I thought I could when I started writing, but by the time I got here, I can't.